### ISSUE TITLE
Too few arguments error when creating an empty TypedDict

### ISSUE THREAD

**Issue by jzazo** [python/mypy#13705]
**Bug Report**
When creating a `TypedDict` only `typename` is required, but mypy issues an error if no fields are specified.

**To Reproduce**
```
from typing import TypedDict

EmptyDict = TypedDict("EmptyDict")  # mypy error:  Too few arguments for TypedDict()  [misc]
EmptyDict.__annotations__  # {}
EmptyDict()  # {}
```
However, the above construct is valid in python.

**Expected Behavior**
I would expect no error is displayed.

**Your Environment**
- Mypy version used: 0.971
- Python version used: 3.10.4
- Operating system and version: Ubuntu 22.04

**Comment by erictraut** [comment]
The following form is equivalent and works fine in mypy:
```python
class EmptyDict(TypedDict): pass
```

**Comment by jzazo** [comment]
Thanks, I will use that in the meantime.

**Comment by hauntsaninja** [comment]
You can also use `EmptyDict = TypedDict("EmptyDict", {})`. The whole point of mypy is that it rejects some things that the runtime accepts ;-)
This feels like a lint that could help catch some stupid mistakes and it's easy enough to be explicit, so closing this as "as-designed"
